Src/Particle/AMReX_ParticleTransformation.H

```diff
@@ -119,7 +123,11 @@ void swapParticle (const ParticleTileData<T_ParticleType, NAR, NAI>& dst,
     AMREX_ASSERT(dst.m_num_runtime_real == src.m_num_runtime_real);
     AMREX_ASSERT(dst.m_num_runtime_int  == src.m_num_runtime_int );
 
-    amrex::Swap(src.m_aos[src_i], dst.m_aos[dst_i]);
+    if constexpr(!T_ParticleType::is_soa_particle) {
+        amrex::Swap(src.m_aos[src_i], dst.m_aos[dst_i]);
+    } else {
+        amrex::Swap(src.m_idcpu[src_i], dst.m_idcpu[dst_i]);
+    }
```

Comment:
I find the opposite order clearer, avoiding the negation:
```suggestion
    if constexpr(T_ParticleType::is_soa_particle) {
        amrex::Swap(src.m_idcpu[src_i], dst.m_idcpu[dst_i]);
    } else {
        amrex::Swap(src.m_aos[src_i], dst.m_aos[dst_i]);
    }
```